app-editors/visual-studio-code-bin/visual-studio-code-bin-1.47.3.ebuild

```diff
@@ -0,0 +1,45 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit desktop eutils pax-utils
+
+DESCRIPTION="Multiplatform Visual Studio Code from Microsoft"
+HOMEPAGE="https://code.visualstudio.com"
+SRC_URI="https://update.code.visualstudio.com/${PV}/linux-x64/stable -> ${P}.tar.gz"
+
+RESTRICT="mirror strip"
+
+LICENSE="MIT Microsoft-VSCode"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
+
+DEPEND="
+	>=media-libs/libpng-1.6.35
+	>=x11-libs/gtk+-3.24.1:3
+	x11-libs/cairo"
+
+RDEPEND="${DEPEND}"
+
+pkg_setup() {
+	S="${WORKDIR}/VSCode-linux-x64"
```

Comment:
We should set S in global scope, right below SRC_URI or right above the first phase definition. So we can get rid of pkg_setup here.